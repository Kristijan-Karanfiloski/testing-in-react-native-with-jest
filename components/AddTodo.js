import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  Button,
  Pressable,
} from "react-native";
import { useState } from "react";

const AddTodo = () => {
  ///////////////////////////////////////////////////////////////////////////////////////////////////

  // const numbers = [1, 2, 3, 4, 5];
  //
  // const increment = (number) => number + 1;
  // const double = (number) => 2 * number;
  // const square = (number) => number * number;
  //
  // const words = ["hello", "world"];
  //
  // const reverse = (string) => [...string].reverse().join();
  // const reversedWordsArray = words.map(reverse);
  //
  // const numbersIncremented = numbers.map(increment);
  //
  // console.log("Numbers incremented :", numbersIncremented);
  // console.log("REVERSED WORDS ARRAY :", reversedWordsArray);

  ////////////////////////////////////////////////////////////////////////////////////////////////
  const [todoItem, setTodoItem] = useState([
    { id: 1, title: "Buy Milk", dueDate: "'2023-10-28'", completed: true },
    { id: 2, title: "Buy Bread", dueDate: "'2023-10-29'", completed: false },
    {
      id: 3,
      title: "Go to the cinema",
      dueDate: "'2023-10-30'",
      completed: true,
    },
  ]);

  const [addNewTodoItemTitle, setAddNewTodoItemTitle] = useState("");
  const [addNewTodoDueDate, setAddNewTodoDueDate] = useState("");
  const [errorMessage, setErrorMessage] = useState();

  // function greet(message) {
  //   function sayHello() {
  //     return `Hello ${message}`;
  //   }
  //
  //   return sayHello();
  // }
  //
  // console.log(greet("fucking kiko"));

  // const addSumOfTwoNumbers = (a, b) => {
  //   console.log("RESULT :", a + b);
  // };
  //
  // const a = 23;
  // const c = 32;
  //
  // addSumOfTwoNumbers(a, 23);

  //! THIS IS IF I WANT TO USE THE FUNCTION AS A HELLPER
  const deleteTodoItem = (id) => {
    return (kiko) => {
      console.log(
        "TODO ITEM FROM THE FUNCTION DELETE TODO ITEM FUNCTION :",
        kiko
      );
      return kiko.id !== id;
    };
  };

  // const onPressDeleteTodoItem = (id) => {
  //   setTodoItem(todoItem.filter(deleteTodoItem(id)));
  // };
  const onPressDeleteTodoItem = (id) => {
    console.log("ID FROM THE OUTER FUNCTION :", id);
    // setTodoItem(todoItem.filter(deleteTodoItem(id)));
    setTodoItem(
      todoItem.filter((item) => {
        console.log("ITEM  FROM INNER FUNCTION :", item);
        console.log("ITEM . ID FROM INNER FUNCTION :", item.id);
        console.log("ID FROM INNER FUNCTION :", id);

        return item.id !== id;
      })
    );
  };

  // const words = ["spr", "elite"];
  //
  // const removeWord = words.filter((word) => word.length > 3);

  // console.log("FILTER WORDS :", removeWord);

  // const newTodoItem = [
  //   [
  //     ...todoItem,
  //     { id: todoItem.length + 1, title: "Buy Pool", dueDate: "'2023-10-27'" },
  //   ],
  // ];

  // console.log("ADDED ITEM TO THE ARRAY :", newTodoItem);
  // console.log("THE TODO ITEMS ARRAY :", todoItem);

  const onPressAddTodoItem = () => {
    if (addNewTodoItemTitle === "") {
      console.log("Cant be empty");
      return;
    }

    setTodoItem([
      ...todoItem,
      {
        id: todoItem.length + 1,
        title: addNewTodoItemTitle,
        dueDate: addNewTodoDueDate,
      },
    ]);
    setAddNewTodoItemTitle("");
    setAddNewTodoDueDate("");
    // setTodoItem();
  };

  return (
    <>
      <View style={styles.container}>
        <Text style={styles.header}>Todo Form</Text>
        <View style={styles.form}>
          <TextInput
            style={styles.input}
            placeholder="New Todo"
            onChangeText={setAddNewTodoItemTitle}
          />
          <TextInput
            style={styles.input}
            placeholder="Due Date"
            onChangeText={setAddNewTodoDueDate}
          />
          <Text>{addNewTodoItemTitle}</Text>
          <Text>{addNewTodoDueDate}</Text>
          {!setAddNewTodoItemTitle && <Text>Cant be empty</Text>}
          <TouchableOpacity style={styles.button} onPress={onPressAddTodoItem}>
            <Text>Add new todo</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.form}>
          <Text>Todo List:</Text>
          <View style={styles.listContainer}>
            {todoItem.map((item) => {
              // console.log("ITEM : ", item);
              return (
                <ul key={item.id} style={styles.listItemContainer}>
                  {/*<Text>{item.title}</Text>*/}
                  {/*<Text>{item.dueDate}</Text>*/}
                  {/*<Text>{item.completed === true ? "done" : "to be done"}</Text>*/}
                  {/*<li>{item.title}</li>*/}
                  <li>
                    {` ${item.title},${item.dueDate} : ${
                      item.completed === true ? "done" : "to be done"
                    }`}
                    {/*<Button title="delete" onPress={onPressDeleteTodoItem} />*/}
                    <Pressable onPress={() => onPressDeleteTodoItem(item.id)}>
                      <Text>Delete</Text>
                    </Pressable>
                  </li>
                </ul>
              );
            })}
          </View>
        </View>
      </View>
    </>
  );
};

export default AddTodo;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "yellow",
    alignItems: "center",
    borderWidth: 7,
    borderColor: "blue",
  },
  header: {
    color: "red",
    marginTop: 10,
  },
  form: {
    marginTop: 16,
    borderWidth: 3,
    borderColor: "blue",
    gap: 1,
    padding: 10,
  },
  input: {
    borderWidth: 1,
    padding: 10,
    marginTop: 16,
  },
  button: {
    marginTop: 20,
    alignItems: "center",
    backgroundColor: "#DDDDDD",
    padding: 10,
  },
  listContainer: {
    padding: 10,
  },
  listItemContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    borderWidth: 3,
    borderColor: "blue",
  },
});
