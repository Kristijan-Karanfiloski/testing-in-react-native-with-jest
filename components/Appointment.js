import React from "react";
import { Text } from "react-native";

const Appointment = () => {
  // const person = {
  //   customer: {
  //     firstName: "Ashley",
  //     lastName: "Jones",
  //     phoneNumber: "(123)555-0123",
  //   },
  //   stylist: "Jay Spears",
  //   startsAt: "2023-02-23 09:30",
  //   service: "Cut",
  //   notes: "",
  // };

  // const firstName = person.customer.firstName;
  const firstName2 = "Ashley";

  return (
    <>
      {/*<Text>{person.customer.firstName}</Text>*/}
      <Text>{firstName2}</Text>
    </>
  );
};

export default Appointment;
