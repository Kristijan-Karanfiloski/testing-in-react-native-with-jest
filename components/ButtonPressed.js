import { Text, TouchableOpacity } from "react-native";

const ButtonPressed = ({ handlePress }) => {
  // const handlePress = () => {
  //   console.log("Button Pressed!");
  // };

  return (
    <TouchableOpacity onPress={handlePress}>
      <Text>Button</Text>
    </TouchableOpacity>
  );
};

export default ButtonPressed;
