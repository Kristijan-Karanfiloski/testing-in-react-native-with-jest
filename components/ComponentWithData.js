import {
  FlatList,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

const data = [
  { id: "1", text: "Text 1" },
  { id: "2", text: "Text 2" },
  { id: "3", text: "Text 3" },
];

const ComponentWithData = ({ onItemPress }) => {
  const Item = (props) => {
    return (
      <View>
        <Text>{props.title}</Text>
      </View>
    );
  };
  const onPressItem = (id) => {
    // console.log(typeof id);
    onItemPress(id);
  };
  const renderItem = ({ item }) => (
    <TouchableOpacity key={item.id} onPress={() => onPressItem(item.id)}>
      <Item title={item.text} />
    </TouchableOpacity>
  );

  return (
    <View style={styles.container}>
      <FlatList
        testID="flat-list"
        data={data}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
      />
    </View>
  );
};

export default ComponentWithData;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: '#fff',
    backgroundColor: "yellow",
    alignItems: "center",
    justifyContent: "center",
    borderColor: "blue",
    borderWidth: 7,
    // height: "50vh",
  },
});
