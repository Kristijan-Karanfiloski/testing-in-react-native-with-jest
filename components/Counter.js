import React, { useState } from "react";
import { Button, StyleSheet, Text, View } from "react-native";

const Counter = () => {
  const [counter, setCounter] = useState(0);

  const incrementCounter = () => {
    setCounter((prevCounter) => prevCounter + 2);
  };
  const decrementCounter = () => {
    setCounter((prevCounter) => {
      if (prevCounter === 0) return 0;
      return prevCounter - 1;
    });
  };

  return (
    <View style={styles.container}>
      <Button title="Decrement" onPress={decrementCounter} />
      <Text testID="counter">{counter}</Text>
      <Button testID="increment" title="Increment" onPress={incrementCounter} />
    </View>
  );
};

export default Counter;

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    // backgroundColor: '#fff',
    backgroundColor: "yellow",
    alignItems: "center",
    justifyContent: "center",
    border: "7px solid blue",
    // height: "50vh",
  },
});
