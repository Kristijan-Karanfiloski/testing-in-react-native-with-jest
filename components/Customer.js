import React from "react";
import { View } from "react-native";
import Product from "./Product";

const product = {
  name: "Bread",
  quantity: 5,
  price: "$50",
};

const quantity = product.quantity;

// console.log("Quantity: :", quantity);

const transformToAString = quantity.toString();

// console.log("TransformToAString :", transformToAString);
// console.log(typeof transformToAString);

const Customer = () => {
  return (
    <View>
      <Product product={product} />
    </View>
  );
};

export default Customer;
