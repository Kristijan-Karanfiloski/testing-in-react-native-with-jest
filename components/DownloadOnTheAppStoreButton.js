import React from "react";
import { Pressable, StyleSheet } from "react-native";

const DownloadOnTheAppStoreButton = () => {
  return <Pressable style={styles.image}>PRess</Pressable>;
};

export default DownloadOnTheAppStoreButton;

const styles = StyleSheet.create({
  image: {
    flex: 1,
    width: 120,
    height: 40,
    backgroundColor: "purple",
    justifyContent: "center",
    alignItems: "center",
  },
});
