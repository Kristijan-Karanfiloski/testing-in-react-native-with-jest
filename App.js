import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import Multiplier from "./components/Multiplier";
import UserForm from "./components/UserForm";
import { useState } from "react";
// import UserList from "./components/UserList";
import ComponentWithData from "./components/ComponentWithData";
import ButtonPressed from "./components/ButtonPressed";
import FetchingDataComponent from "./components/FetchingDataComponent";
import Product from "./components/Product";
import Customer from "./components/Customer";
import Counter from "./components/Counter";
import IsPalindrome from "./components/IsPalindrome";

export default function App() {
  const [users, setUsers] = useState([]);

  const onAddUser = (user) => {
    setUsers([...users, user]);
    console.log("USER FROM APP :", users);
  };

  const handlePress = () => {
    console.log("Button Pressed!");
  };

  return (
    <View style={styles.wrapper}>
      {/*<Text>Open up App.js to start working on your app!</Text>*/}
      {/*<Multiplier />*/}
      {/*<StatusBar style="auto" />*/}

      {/*<View style={styles.container}>*/}
      {/*<UserForm onAddUser={onAddUser} />*/}
      {/*<UserList users={users} />*/}
      {/*</View>*/}
      {/*<Product />*/}
      {/*<Customer />*/}
      <ComponentWithData />
      <FetchingDataComponent />
      <Counter />
      <IsPalindrome />
      {/*<UserList users={users} />*/}
      {/*<ButtonPressed handlePress={handlePress} />*/}
    </View>
  );
}

const styles = StyleSheet.create({
  wrapper: {
    display: "flex",
    flexDirection: "column",
    flex: 1,
  },
  container: {
    // flex: 1,
    // backgroundColor: '#fff',
    backgroundColor: "yellow",
    alignItems: "center",
    justifyContent: "center",
    border: "7px solid blue",
    // height: "50vh",
  },
});
