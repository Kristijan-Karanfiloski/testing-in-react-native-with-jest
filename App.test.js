import renderer from "react-test-renderer";
import App from "./App";
import { addTwoNumbers } from "./components/utils";

//!This begins a new test suite (a collection of tests) for the App component.
// describe("App", () => {
//   it("has 1 child", () => {
//     ////////////////////////////////////////////////////////////////////////////////////////////////
//     //!This line creates a new instance of the App component, renders it to a JSON object, and assigns this object
//     //! to a variable named tree.
//     const tree = renderer.create(<App />).toJSON();
//
//     ////////////////////////////////////////////////////////////////////////////////////////////////
//
//     //!This is an assertion that checks if the tree object has exactly one child. If it does, this test will pass;
//     //! if it doesn't, this test will fail.
//     expect(tree.children.length).toBe(1);
//   });
//
//   it("renders correctly", () => {
//     const tree = renderer.create(<App />).toJSON();
//
//     //!    This is an assertion that checks if the tree object matches a previously saved snapshot. If it does, this
//     //!  test will pass; if it doesn't, this test will fail, indicating that the App component's rendered output has
//     //! changed.
//
//     expect(tree).toMatchSnapshot();
//   });
// });

describe("testing the add function from utils", () => {
  it("should add two numbers correctly", () => {
    expect(addTwoNumbers(2, 3)).toEqual(5);
    expect(addTwoNumbers(-2, 3)).toBe(1);
    expect(addTwoNumbers(-2, -3)).toBe(-5);
  });
});
