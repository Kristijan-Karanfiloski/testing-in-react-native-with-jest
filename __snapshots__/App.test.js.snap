// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "borderColor": "blue",
        "borderWidth": 7,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": "1",
            "text": "Text 1",
          },
          {
            "id": "2",
            "text": "Text 2",
          },
          {
            "id": "3",
            "text": "Text 3",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="flat-list"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View>
              <Text>
                Text 1
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View>
              <Text>
                Text 2
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View>
              <Text>
                Text 3
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "border": "7px solid blue",
        "justifyContent": "center",
      }
    }
  >
    <Text>
      No coffee found
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "border": "7px solid blue",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Decrement
        </Text>
      </View>
    </View>
    <Text
      testID="counter"
    >
      0
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="increment"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Increment
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "yellow",
        "borderColor": "blue",
        "borderWidth": 7,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          check if palindrome
        </Text>
      </View>
    </View>
    <Text>
      Fuck
    </Text>
  </View>
</View>
`;
